### PR TITLE
Included COGify command, improved metadata included in STAC

### LINKED ISSUE #3: Package rename
The package should be renamed since it is a subset of SPOT data.

This dataset is from the legacy geobase project. It now is referenced through https://open.canada.ca/data/en/dataset/d799c202-603d-4e5c-b1eb-d058803f80f9

Suggest the name change to `nrcan_spot_ortho`

### PR DESCRIPTION
**Pull request by jamesvrt**
- Included a `cogify` command
- Included `proj` and `eo` extensions within the STAC for improved metadata
- Fixed the `pystac` IO to correctly allow reading and writing to AWS S3

### CODE REVIEW

**Comment by jamesvrt** [comment]
Includes #3

File: README.md [hunk 1]
```diff
@@ -1,15 +1,20 @@
-# stactools-spot
+# stactools-nrcan_spot_ortho
```
Comment (gadomski) [inline README.md]:
```suggestion
# stactools-nrcan-spot-ortho
```

Think this should be hyphens not underscores in multiple places including here.

File: README.md [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
+The root href can be a local or S3 path. The default catalog type is `pystac.CatalogType.ABSOLUTE_PUBLISHED`.
+
+The STAC catalog created contains assets with hrefs pointing to zipped imagery on the Geobase FTP. These can be be downloaded, unzipped and converted to COGs with:
+```
+stac nrcan_spot_ortho cogify-assets [catalog path] -d [COG directory]
+```
+
+By default the COGs will be written into the same directory's as their respective STAC items, and will __not__ overwrite existing COGs (use `--overwrite` to do so). The STAC will be updated with the COG assets during this process.
+
+A complete orthorectified SPOT 4 and 5 STAC, including COGs, can be found [here](). An API for this STAC can be found [here]().
```
Comment (gadomski) [inline README.md]: Should be filled in or not included?

File: setup.cfg [hunk 3]
```diff
@@ -28,7 +28,10 @@ package_dir =
     = src
 packages = find_namespace:
 install_requires =
-    stactools ~= 0.1
+    stactools ~= 0.2.1a1
```
Comment (gadomski) [inline setup.cfg]:
```suggestion
    stactools == 0.2.1
```

File: src/stactools/nrcan_spot_ortho/cog.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+    if (not overwrite) and (output_path in existing_cog_paths):
+        print(f"Skipping {os.path.basename(output_path)}, already COGified.")
+
+    elif parsed.scheme == "s3":
+        with TemporaryDirectory() as tmp_dir:
+            tmp_path = os.path.join(tmp_dir, os.path.basename(output_path))
+            failure = call([
+                'gdal_translate', '-of', 'COG', '-co', 'compress=deflate',
+                input_path, tmp_path
+            ])
```
Comment (gadomski) [inline src/stactools/nrcan_spot_ortho/cog.py:35]: Not required, but there is a `cogify` command in `stactools.core.utils`: https://github.com/stac-utils/stactools/blob/a2e541e18d1d50689c5a0a5bb224bccbcbdf6819/src/stactools/core/utils/convert.py#L8-L20

File: src/stactools/nrcan_spot_ortho/utils.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            paths += [f"{directory}/{d['Key']}" for d in page['Contents'] 
+            if d['Key'][-len(ending):] == ending]
+        return paths
+
+    else:
+        return glob.glob(f"{directory}{os.sep}**{os.sep}*{ending}",
+                         recursive=True)
+
+
+def file_exists(path, paths_s3):
```
Comment (gadomski) [inline src/stactools/nrcan_spot_ortho/utils.py]: There is a new `href_exists` function in `stactools` that does something similar, but this looks good too: https://github.com/stac-utils/stactools/blob/main/src/stactools/core/utils/__init__.py#L16-L23.

**Review by gadomski — CHANGES_REQUESTED** [review]
The package name has hypens not underscores, otherwise looks alright!

File: src/stactools/nrcan_spot_ortho/utils.py (hunk 5 above)
Comment (jamesvrt) [reply src/stactools/nrcan_spot_ortho/utils.py]: Noted, I'll use that for future implementations. Turns out I don't even use this function after all so I'll remove it. I collect all existing paths upfront with `get_existing_paths` then compare with that list. Made s3 interactions faster!

**Review by jamesvrt — COMMENTED** [review]
Requested changes incorporated

**Comment by jamesvrt** [comment]
I'm not sure how to resolve the conflicting files. I may not have sufficient permission.

**Review by gadomski — APPROVED** [review]

**Comment by gadomski** [comment]
@jamesvrt I was able to successfully rebase your branch onto **main** (e.g. for you that would be `git rebase origin/main` or something similar). If you do that and then force push your rebased branch back up, that should take care of the conflicts.

**Comment by jamesvrt** [comment]
Slowly getting the hang of this ol' git thing :turtle: Thanks again for your patience

**Comment by gadomski** [comment]
No problem at all! Thanks for iterating. One more change that might need to happen is that PySTAC removed Python 3.6 support, so you'll need to remove 3.6 from the CI here: https://github.com/jamesvrt/nrcan_spot_ortho/blob/main/.github/workflows/continuous-integration.yml#L28.

**Pull request merged**